File: linked lists/160. Intersection of Two Linked Lists/getIntersectionNode.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        # Convert list A to a list of nodes
        node = headA
        lstA = []
        while node:
            lstA.append(node)
            node = node.next
        
        # Convert list B to a list of nodes
        node = headB
        lstB = []
        while node:
            lstB.append(node)
            node = node.next
            
        lenA, lenB = len(lstA), len(lstB)
        
        # Adjust the lists by skipping the difference in lengths
        diff = abs(lenA - lenB)
        if lenA > lenB:
            # Skip extra nodes at the beginning in A
            lstA = lstA[diff:]
        else:
            # Skip extra nodes at the beginning in B
            lstB = lstB[diff:]
        
        # Compare nodes starting from the first node
        for i in range(min(lenA, lenB)):
            # If the intersection is found
            if lstA[i] == lstB[i]:
                # return the intersection node
                return lstA[i]
        
        return None

File: linked lists/160. Intersection of Two Linked Lists/getIntersectionNode.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    let node = headA;
    let lstA = [];
    while (node) {
        lstA.push(node);
        node = node.next;
    }

    node = headB;
    let lstB = [];
    while (node) {
        lstB.push(node);
        node = node.next;
    }

    const lenA = lstA.length;
    const lenB = lstB.length;

    const diff = Math.abs(lenA - lenB);

    if (lenA > lenB) {
        lstA.splice(0, diff);
    } else {
        lstB.splice(0, diff);
    }

    for (let i = 0; i < Math.min(lenA, lenB); i++) {
        if (lstA[i] == lstB[i]) {
            return lstA[i];
        }
    }

    return null;
};